EIPS/eip-2456.md

```diff
@@ -0,0 +1,166 @@
+---
+eip: 2456
+title: Time Based Upgrade Transitions
+author: Danno Ferrin (@shemnon)
+discussions-to: https://ethereum-magicians.org/t/time-based-upgrade-transitions/3902
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-01-02
+---
+
+## Simple Summary
+
+A process to specify network upgrades relative to a point in time instead of a fixed block number.
+
+## Abstract
+
+Instead of assigning network upgrade transitions to occur on `TRANSITION_BLOCK` ahead of time a
+`TRANSITION_TIME` is chosen. On the second "round" block after the `TRANSITION_TIME` occurs the
+transition to the network upgrade will occur. Meta-EIPs will list the TIME and later be updated to
+also include the historical BLOCK when the upgrade transitioned.
+
+## Motivation
+
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+
+For the Muir Glacier and Istanbul upgrades there were a total of at least 4 times when the planned
+upgrade occurred well away from the intended date. For Ropsten the Istanbul upgrade occurred earlier
+because of the sudden addition of hashing power, and for Muir Glacier it happened much later because
+of the removal of that hashing power. For Mainnet the Istanbul transition was days later and the
+Muir Glacier transition was days sooner because of the unpredictable impact of the Difficulty Bomb.
+
+Such unpredictability is bad for node operators such as exchanges and block explorers. Particularly
+an earlier fork may disrupt their maintenance plans. A much shorter window of unpredictability would
+benefit node operators and developers monitoring the upgrade.
+
+## Specification
+
+A 'Transition Eligible Block' is defined as a block where the number of blocks preceding the block
+is evenly divisible by the `TRANSITION_INCREMENT`, i.e.
+`(block.number % TRANSITION_INCREMENT) == 0`. For Mainnet the `TRANSITION_INCREMENT` will be 1000.
+For testnets a `TRANSITION_INCREMENT` of 1024 is recommended.
+
+A specific `TRANSITION_TIME` will be chosen for the network upgrade to activate.
+
+An upgrade will activate at a Transition Eligible Block if all of the following are true:
+
+- The upgrade has not activated already.
+- The timestamp of the block is on or after the `TRANSITION_TIME`.
+- The previous Transition Eligible Block was on or after the `TRANSITION_TIME`.
```

Comment:
This is ambiguous, since and I'm wondering if there's some typo...? 
Because if the timestamp of `N` is on `TRANSITION_TIME`, then the timestamp of `N-1` cannot also be `on` `TRANSITION TIME`. So as I read it, we can totally skip the second bullet?

Reply:
I'll tighten up the wording on the second bullet point.  "the block" of the second bullet point needs to be a Transition Eligible Block.  So point 2 is looking at block `n` where `(n % 1000) == 0` and point 3 is looking at block `n - TRANSISION_INCREMENT` and hence `((n - TRANSITION_INCREMENT) % 1000 == 0)`